ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -149,41 +157,130 @@ public static String getCommandsStreamName() {
     this.streamedQueryResource =
         Objects.requireNonNull(streamedQueryResource, "streamedQueryResource");
     this.ksqlResource = Objects.requireNonNull(ksqlResource, "ksqlResource");
-    this.commandQueue = Objects.requireNonNull(commandQueue, "commandQueue");
-
+    this.commandStore = Objects.requireNonNull(commandStore, "commandStore");
+    this.serverState = Objects.requireNonNull(serverState, "serverState");
+    this.processingLogContext = Objects.requireNonNull(
+        processingLogContext,
+        "processingLogContext");
+    this.preconditions = Objects.requireNonNull(preconditions, "preconditions");
     this.versionCheckerAgent =
         Objects.requireNonNull(versionCheckerAgent, "versionCheckerAgent");
-    this.serverInfo = new ServerInfo(
-        Version.getVersion(),
-        getKafkaClusterId(serviceContext),
-        ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG));
 
     this.commandRunnerThread = new Thread(commandRunner, "CommandRunner");
   }
 
   @Override
   public void setupResources(final Configurable<?> config, final KsqlRestConfig appConfig) {
     config.register(rootDocument);
-    config.register(new ServerInfoResource(serverInfo));
+    config.register(new ServerInfoResource(serviceContext, ksqlConfig));
     config.register(statusResource);
     config.register(ksqlResource);
     config.register(streamedQueryResource);
     config.register(new KsqlExceptionMapper());
+    config.register(new ServerStateDynamicBinding(serverState));
   }
 
   @Override
   public void start() throws Exception {
-    commandRunnerThread.start();
     super.start();
+    startKsql();
+    commandRunnerThread.start();
     final Properties metricsProperties = new Properties();
     metricsProperties.putAll(getConfiguration().getOriginals());
     if (versionCheckerAgent != null) {
       versionCheckerAgent.start(KsqlModuleType.SERVER, metricsProperties);
     }
-
     displayWelcomeMessage();
   }
 
+  @VisibleForTesting
+  void startKsql() {
```

Comment:
I plan to refactor this into a separate class in 5.3, so that we decouple managing process start up from setting up the rest server. Wanted to avoid doing that on this branch since this stuff has changed a lot recently with the topic management and rbac work.